src/metabase/util/honeysql_extensions.clj

```diff
@@ -214,19 +214,25 @@
   {:added "0.39.0"}
   (::database-type type-info))
 
+(defn database-type
+  "Returns the `database-type` from the type-info of `honeysql-form` if present.
+   Otherwise, returns `nil`."
+  [honeysql-form]
+  (some-> honeysql-form type-info type-info->db-type))
```

Comment:
Just added a helper function which is called by `datetime-diff-check-args`.
https://github.com/metabase/metabase/pull/27237/files#diff-de45e1639c2976212706ab34736e8fbca40b497026adca37033da380fca04738R684

Reply:
👍 I don't know why we didn't already have this function to be honest